### PR TITLE
Refactors canister damage calculation

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
Title
https://github.com/tgstation/tgstation/pull/66460 without the balance changes so this pr can go faster. Still want to do the balance changes, but that can wait.

## Why It's Good For The Game
Codeing

## Changelog
:cl:
refactor: Changed how canister calculate their overpressure/overtemp damage. No gameplay changes expected.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -596,15 +590,8 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
...
 			air_update_turf(FALSE, FALSE)
 
-	var/our_pressure = air_contents.return_pressure()
-	var/our_temperature = air_contents.return_temperature()
-
-	///function used to check the limit of the canisters and also set the amount of damage that the canister can receive, if the heat and pressure are way higher than the limit the more damage will be done
-	if(!protected_contents && (our_temperature > heat_limit || our_pressure > pressure_limit))
-		take_damage(clamp((our_temperature/heat_limit) * (our_pressure/pressure_limit), 5, 50), BURN, 0)
-		excited = TRUE
+	excited = take_atmos_damage() // A bit different than other atmos devices it seems. Wont stop if currently taking damage.
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/portable/canister.dm:593]: would prefer an if here so it's not set to false ever, more future proof

File: code/modules/atmospherics/machinery/portable/portable_atmospherics.dm [hunk 2]
```diff
@@ -17,6 +19,11 @@
 	///Used to track if anything of note has happen while running process_atmos()
 	var/excited = TRUE
 
+	/// Max amount of heat allowed inside the machine before it starts to melt. -1 is special value meaning we wont check.
+	var/temp_limit = 5000
+	/// Max amount of pressure allowed inside of the canister before it starts to break. -1 is special value meaning we wont check.
+	var/pressure_limit = 50000
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/portable/portable_atmospherics.dm:25]: we won't check vs we are immune

Comment (LemonInTheDark) [reply code/modules/atmospherics/machinery/portable/portable_atmospherics.dm:25]: also is it -1 or 0

File: code/modules/atmospherics/machinery/portable/portable_atmospherics.dm [hunk 3]
```diff
@@ -49,6 +56,29 @@
...
+		taking_damage = temp_damage > 1
+
+	if(pressure_limit != PORTABLE_ATMOS_IGNORE_ATMOS_LIMIT)
+		pressure_damage = air_contents.return_pressure() / pressure_limit
+		taking_damage = taking_damage || pressure_damage > 1
+
+	if(!taking_damage)
+		return FALSE
+	
+	take_damage(clamp(temp_damage * pressure_damage, 5, 50), BURN, 0)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/portable/portable_atmospherics.dm:79]: clean

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/portable/portable_atmospherics.dm (hunk 2 above)
Comment (vincentiusvin) [reply code/modules/atmospherics/machinery/portable/portable_atmospherics.dm:25]: -1 earlier but moved it to 0. Forgot to update.

**Review by LemonInTheDark — APPROVED** [review]

**Comment by TemporalOroboros** [comment]
You might want to extend this to the stationary tanks because right now they just take a flat 0.1 damage/tick when over the pressure threshold. If I'm right this makes alien alloy tanks not actually capable of rupturing on their own because of the regen they get.

**Comment by vincentiusvin** [comment]
They aren't portables it seems, guess someone else can do that. Might be worth an issue report.

**Pull request merged**